### PR TITLE
feat(HighContrast): Expose the XAML HighContrastChanged event to JS

### PR DESCRIPTION
**Pull request by antonkh**
The event is exposed as highContrastDidChange in AccessibilityInfo. For now
it only tells if high contrast is enabled or disabled in system settings. Later it
can tell what are the chosen high contrast colors: Windows 10 allows to create
custom high contrast themes with user chosen colors.

### CODE REVIEW

File: ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    // TODO: It would be better to have a sync GethIsHighContrast,
+                    // but this is not supported by the framework at the moment.
+                    { "initialHighContrast", _accessibility.HighContrast },
+                };
+            }
+        }
+
+        public override void Initialize()
+        {
+            _accessibility.HighContrastChanged += OnHighContrastChanged;
```
Comment (rigdern) [inline ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs]: >_accessibility.HighContrastChanged += OnHighContrastChanged; [](start = 12, length = 60)

Should we do this in `Initialize` or in the constructor?

File: ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        private readonly AccessibilitySettings _accessibility = new AccessibilitySettings();
+        private readonly UISettings _settings = new UISettings();
+
+        private string GetRgbaString(UIElementType type)
+        {
+            var color = _settings.UIElementColor(type);
+            return "rgba(" + color.R + "," + color.G + "," + color.B + "," + color.A + ")";
+        }
+
+        private IDictionary<string, string> GetHighContrastColors()
```
Comment (rigdern) [inline ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs]: >private IDictionary<string, string> GetHighContrastColors() [](start = 8, length = 59)

Maybe put a TODO about how we want to expose these colors to JavaScript in the future

File: Libraries/Components/AccessibilityInfo/AccessibilityInfo.windows.js [hunk 3]
```diff
@@ -11,32 +11,55 @@
...
 
 var AccessibilityInfo = {
 
-  fetch: function(): Promise<*> {
+  initialHighContrast: RCTAccessibilityInfo.initialHighContrast,
+  
+  fetch: function(): Promise {
     return new Promise((resolve, reject) => {
-      reject('AccessibilityInfo is not supported on this platform.');
+      resolve(false);
```
Comment (rigdern) [inline Libraries/Components/AccessibilityInfo/AccessibilityInfo.windows.js]: >resolve(false); [](start = 6, length = 15)

Can you put a comment about how UWP doesn't provide anyway to query whether or not a screen reader is being used?

File: ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs (hunk 1 above)
Comment (antonkh) [reply ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs]: Eric suggested to use `Initialize`.

File: ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs (hunk 2 above)
Comment (antonkh) [reply ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs]: Added

File: Libraries/Components/AccessibilityInfo/AccessibilityInfo.windows.js (hunk 3 above)
Comment (antonkh) [reply Libraries/Components/AccessibilityInfo/AccessibilityInfo.windows.js]: Done

File: ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs (hunk 1 above)
Comment (rigdern) [reply ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs]: I see. What's the reasoning behind preferring `Initialize` over the constructor? When does `Initialize` get called relative to `Constants`?

File: ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        private readonly AccessibilitySettings _accessibility = new AccessibilitySettings();
+        private readonly UISettings _settings = new UISettings();
+
+        private string GetRgbaString(UIElementType type)
+        {
+            var color = _settings.UIElementColor(type);
+            return "rgba(" + color.R + "," + color.G + "," + color.B + "," + color.A + ")";
+        }
+
+        // TODO: These are the 8 high contrast colors in Windows. Eventually they will be exposed to JS.
```
Comment (matthargett) [inline ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs]: can you file an issue for this TODO item?

File: ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs (hunk 1 above)
Comment (rozele) [reply ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs]: Constructor is called before the JS bundle is loaded, so you could potentially notify event to JS before anything there to listen. If you call after Initialize, bundle load operation has already been queued.

---
In reply to: [144430206](https://github.com/Microsoft/react-native-windows/pull/1443#discussion_r144430206) [](ancestors = 144430206)

Comment (rigdern) [reply ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs]: Firing an event before anything is listening sounds pretty harmless.

So it sounds like it doesn't matter too much whether you register for the event in `Initialize` or in the constructor. Do you agree?

---
In reply to: [145428282](https://github.com/Microsoft/react-native-windows/pull/1443#discussion_r145428282) [](ancestors = 145428282,144430206)

File: ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs (hunk 4 above)
Comment (antonkh) [reply ReactWindows/ReactNative/Modules/AccessibilityInfo/AccessibilityInfoModule.cs]: Done: https://github.com/Microsoft/react-native-windows/issues/1459

**Pull request merged**